### ISSUE TITLE
Default to vcpkg instead of pkg-config for easy Windows building

### ISSUE THREAD

**Issue by monax3** [twistedfall/opencv-rust#68]
After a couple days of fighting link errors when trying to use a version of opencv installed via [vcpkg](https://github.com/microsoft/vcpkg) I hacked the build script to use the [vcpkg crate](https://crates.io/crates/vcpkg) to get the opencv paths including header path and was immediately succesful except for having to edit gen_rust.py to add `loop` to the reserved keywords.

Properly implemented, this would allow easy MSVC builds via:
```
vcpkg install opencv4[necessary features here]
cargo build
```

**Comment by twistedfall** [comment]
Hi, will it be possible for you to share the resulting code?

**Comment by monax3** [comment]
This is the main part. I had to rearrange other things non-gracefully to let the probe return the header path instead of defaulting to the bundled ones.
```Rust
    #[cfg(target_env = "msvc")]
    fn probe_vcpkg(pkg_name: &str) -> Result<Self> {
        let opencv = vcpkg::Config::new().find_package(pkg_name)?;
        let libs: Vec<String> = opencv
            .found_libs
            .iter()
            .map(|lib| lib.file_name().unwrap().to_string_lossy().to_string())
            .collect();
        let version = opencv
            .include_paths
            .iter()
            .filter_map(get_version_from_headers)
            .next();
        let libdir = opencv
            .link_paths
            .first()
            .map(|x| x.clone())
            .unwrap_or_else(|| PathBuf::from(""));
        let header_path = opencv
            .include_paths
            .first()
            .map(|x| x.clone())
            .unwrap_or_else(|| PathBuf::from(""));

        Ok(Self {
            pkg_name: pkg_name.to_owned(),
            libs,
            link_paths: opencv.link_paths,
            framework_paths: vec![],
            include_paths: opencv.include_paths,
            header_path: Some(header_path),
            version: version.unwrap_or("0.0.0".to_owned()),
            prefix: PathBuf::from(""),
            libdir: libdir,
        })
    }
```

**Comment by twistedfall** [comment]
Thanks! If you could do the pull request with all of the changes I'd really appreciate it too.

**Comment by monax3** [comment]
I'll try to get it cleaned up

**Comment by twistedfall** [comment]
Even in the "messy" form it would be welcome, there is no need to actually make it merge ready, I just want to take a look at the changes in their wholeness. I don't have any experience with vcpkg so I might miss some minor detail if I don't have the whole changeset.

**Comment by twistedfall** [comment]
Can you please check if `vcpkg` branch still works for you for hassle-free build with vcpkg? I removed the `OPENCV_HEADER_PATH` var btw so if you have set it up in the past it won't have any effect.

**Comment by monax3** [comment]
I can't figure out why at a quick glance, but your version causes link errors without --release

**Comment by twistedfall** [comment]
Can you please share the full build log?
* `cargo clean`
* `cargo build -vv`
And also:
* `cargo clean`
* `cargo build -vv --features "buildtime-bindgen"`

**Comment by twistedfall** [comment]
Hey, @Threep , did you manage to find out what's wrong in the version from `vcpkg` branch compared to yours?

**Comment by monax3** [comment]
My power supply broke down and in the time it took to fix it I forgot about it. Sorry for the delay.

All my builds are with `buildtime-bindgen` but I've either misunderstood how I'm supposed to use the crate or uncovered a bug because this fails:
```
opencv = { git = "https://github.com/twistedfall/opencv-rust", branch = "vcpkg", default-features = false, features = [ "opencv-41", "buildtime-bindgen" ] }`
```
But if I build with `cd opencv-rust && cargo build --no-default-features --features opencv-41,buildtime-bindgen` beforehand, this works with a local copy of the `vcpkg` branch
```
opencv = { path = "opencv-rust", default-features = false, features = [ "opencv-41", "buildtime-bindgen" ] }
```

Also, I missed it in my patch since I only tested with 4.x versions, but the `vcpkg` uses `opencv3` for 3.4 while `opencv` is an allias for `opencv4`

[build-fails.txt](https://github.com/twistedfall/opencv-rust/files/3955248/build-fails.txt)
[build-succeeds1.txt](https://github.com/twistedfall/opencv-rust/files/3955246/build-succeeds1.txt)
[build-succeeds2.txt](https://github.com/twistedfall/opencv-rust/files/3955247/build-succeeds2.txt)

**Comment by twistedfall** [comment]
Thanks for the update. And you're saying that with your initial changes you don't get linking failure on the first try, right? Will it be possible to send the build log of that too? So `cargo clean` and `cargo build -vv` on the branch that you used initially for the pull request.

**Comment by monax3** [comment]
I get the exact same linking failures with my own branch when it's clean, I was running build on my local copy while tinkering with it and didn't realize that affected it

Clean copy directly from git, my fork or yours, or cloned in advance = Link errors on use
Already built local copy = Links without issue

[build-fork.txt](https://github.com/twistedfall/opencv-rust/files/3956360/build-fork.txt)

**Comment by twistedfall** [comment]
Well, vcpkg integration works then, I'll release it. Linking issue on Windows is not new, but your additional info about successful link on the already built crate is interesting, I'll check it out.

**Comment by twistedfall** [comment]
0.27.0 is released

**Comment by twistedfall** [comment]
The linking issue should be fixed in 0.28.0
